debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -89,6 +89,11 @@ public void onEntryChosenForEviction(PreparedStatement statement) {
                 }
             });
 
+    /**
+     * The auto-commit mode to be set on the underlying connection.
+     */
+    private final Boolean autoCommit;
```

Comment:
Do we actually set this anywhere? I.e. is the constructor below ever invoked with a non-null value for this parameter?

Reply:
Yes:
1. https://github.com/debezium/debezium/blob/65b9f15932c255c2e40ebbc687fcd830371e18df/debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorTask.java#L75-L76
2. https://github.com/debezium/debezium/blob/65b9f15932c255c2e40ebbc687fcd830371e18df/debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java#L138-L140
3. https://github.com/debezium/debezium/blob/65b9f15932c255c2e40ebbc687fcd830371e18df/debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java#L390-L395

I still believe that in the long term, each change event source should define its own transaction boundaries (e.g. the entire snapshot, one streaming iteration, a time-limited set of streaming iterations, etc). E.g. reading the MySQL binlog doesn't need any isolation because it's already materialized.

This would not only simplify the API but help avoid issues with long-running transactions by design.